Refactor to hit Openstates API for state senators and hide API keys in secrets.js

tpetersen0308 opened the pull request:
Added API keys to secrets.js to keep them hidden in views.
Added Axios dependency for hitting APIs and switched from Sunlight to Openstates for representative results.

Note: Commit with SHA-1 40b9a33 reflects a google maps API key from the old repo. This API key is not in use.

WCRest commented on app/controllers/campaign.js:
```diff
@@ -1,50 +1,51 @@
+const axios = require("axios");
 var Campaign = require('../models/campaign');
-var sunlight = require('sunlight-congress-api');
+var openstatesApiKey = require("../../conf/secrets.js").openstates_api_key;
```
I second what Sarah says. If we want to hide the openstates key, we need to proxy the openkey data through our webserver. This is a pretty big PR tho, so maybe we create an issue for this and make it high priority so this PR isn't blocked?

WCRest commented:
I think once you add some code that handles receiving back 0 legislators from openstates, this should be good to merge. You might also consider doing a lil survey of openstates data across different municipalities to see how clean the data is. It might not suffice to look for legislators with office.name == "senate office" 

Additionally, much of this PR was tabbing clean-up. This is good! You mentioned that your editor was just rewriting this stuff based on its linting rules. I guess not everyone is configuring their editor to use the .eslintrc.js because a handful of files in the repo don't obey linting rules! I'd suggest running `npm run lint` before committing code.

tpetersen0308 commented:
Just added another commit to select the representative as the first object in the openstates response as opposed to selecting based on having an office with office.name === "senate office". There is error handling in place for a response with 0 results--it wasn't getting triggered in this bug because results were returned, but the office.name in question was null so it just stalled instead. I added a conditional to display the error message not only if there are 0 results, but also if we were unable to retrieve a phone number from the results. Additionally, I'm getting erros when I run `npm run lint`. I'll spend some time trying to debug that, but I think these new commits will help return cleaner rep results.
